src/System.Management.Automation/engine/remoting/common/RemoteSessionNamedPipe.cs

```diff
@@ -108,30 +110,43 @@ internal static class NamedPipeUtils
                 // The starttime is there to prevent another process easily guessing the pipe name
                 // and squatting on it.
                 // There is a limit of 104 characters in total including the temp path to the named pipe file
-                // on non-Windows systems, so we'll convert the starttime to hex and just take the first 8 characters.
+                // on non-Windows systems, so we'll convert the starttime to hex and just take 8 characters.
 #if UNIX
                 .Append(proc.StartTime.ToFileTime().ToString("X8").AsSpan(1, 8))
 #else
                 .Append(proc.StartTime.ToFileTime().ToString(CultureInfo.InvariantCulture))
 #endif
                 .Append('.')
-                .Append(proc.Id.ToString(CultureInfo.InvariantCulture))
-                .Append('.')
+                .Append(proc.Id.ToString(CultureInfo.InvariantCulture));
+
+            int uniqueLength = pipeNameBuilder.Length;
+            pipeNameBuilder.Append('.')
                 .Append(CleanAppDomainNameForPipeName(appDomainName))
                 .Append('.')
                 .Append(proc.ProcessName);
+
+            string pipeName = pipeNameBuilder.ToString();
 #if UNIX
-            int charsToTrim = pipeNameBuilder.Length - MaxNamedPipeNameSize;
+            // .NET Creates named pipes in the tmp path with a special prefix which needs to be included in the max
+            // length calculation.
+            string pipePrefix = Path.Combine(Path.GetTempPath(), "CoreFxPipe_");
+            int remainingPipeLength = Math.Max(MaxNamedPipeNameSize - pipePrefix.Length, 0);
+
+            int charsToTrim = pipeName.Length - remainingPipeLength;
```

Comment:
Note that this is a potential breaking change when connecting to older versions of PowerShell, as a different trimming algorithm is used.

Reply:
Older PowerShell versions fail currently if it exceeds the max length as per https://github.com/PowerShell/PowerShell/issues/16994. It also currently does not handle the location of the pipe as part of the calculation and so pwsh will fail to create the pipe in general if `$TMPDIR` is longer than `/tmp`.